routing/index_graph.hpp

```diff
@@ -89,6 +95,13 @@ class IndexGraph final
     return GetGeometry().GetRoad(segment.GetFeatureId()).GetPoint(segment.GetPointId(front));
   }
 
+  void GetLastPointsForJoint(std::vector<Segment> const & childs, bool isOutgoing, std::vector<uint32_t> & lastPoints);
+  void GetSegmentCandidateForJoint(Segment const & parent, bool isOutgoing, std::vector<Segment> & childs);
+  void ReconstructJointSegment(Segment const & parent, std::vector<Segment> const & firstChilds,
+                               std::vector<uint32_t> const & lastPointIds,
+                               bool isOutgoing, std::vector<JointEdge> & jointEdges,
+                               std::vector<RouteWeight> & parentWeights);
```

Comment:
Комментарий не ней не будет лишним. Особенно к тому, чем заполняются |jointEdges| и |parentWeights|.

Reply:
```cpp
/// \brief Prolongs segments from |parent| to |firstChildren| directions in order to
///        create JointSegments.
/// \param |firstChildren| - vecotor of neigbouring segments from parent.
/// \param |lastPointIds| - vector of the end numbers of road points for |firstChildren|.
/// \param |jointEdges| - the result vector with JointEdges.
/// \param |parentWeights| - see |IndexGraphStarterJoints::GetEdgeList| method about this argument.
///                          Shotly - in case of |isOutgoing| == false, method saves here the weights
///                                   from parent to firstChildren.
```